src/main/java/org/elasticsearch/search/aggregations/AggregationPhase.java

```diff
@@ -120,6 +122,7 @@ public void execute(SearchContext context) throws ElasticsearchException {
                 query = new FilteredQuery(query, searchFilter);
             }
             try {
+                collector.preCollection();
```

Comment:
Is this needed here? Will this not have already been called in the preProcess method?